### DISCUSSION TITLE
발표 자료 준비

### DISCUSSION THREAD

**Discussion by 00TaciTa00** [codeit-fe2-2/taskify#77] [Q&A]
# 이건 꼭 넣고 싶다
- [react-datetime]
  - 날짜와 시간, 포매팅, datetime-local의 기본 아이콘 제거, tailwindcss 적용, 각 내부 컴포넌트에 대한 접근이 모두 용이한 외부 라이브러리를 탐색한 결과
  - tailwindcss를 적극 사용하기로 결정한 현재 프로젝트에서 webkit에 직접 접근하는 것 외의 방법을 탐색했음
- DefaultProfileImage.tsx
  - 프로필 이미지가 없는 유저를 표현할 때 프로필 이미지를 대신할 컴포넌트
  - 배경색을 랜덤으로 하되 모든 유저가 서로를 동일한 색으로 인식해야 했음
  - 각 유저의 닉네임 첫글자를 유니코드로 변환한 뒤, 그 값을 임시로 정해진 컬러 팔레트의 길이로 나눠 그 나머지를 팔레드의 인덱스 값으로 사용.

# 이런 생각 많이 했어요
- 프로젝트를 체계적으로 관리할 수 있게 되니 현재 내가 무엇을 해야하는지, 또 무엇을 할 수 있는지를 알 수 있어서 좋았다.
- datetime 인풋을 작성할 때 외부 라이브러리 없이 해결하겠다고 고집한 시간이 아까웠다. 라이브러리를 찾아 다닐 때도 겉만 화려한 라이브러리를 쓰겠다고 탐색하다 결국 가장 기본적인 것이 제일임을 다시금 깨달았다.
- 그래도 이후 불필요한 스크롤바 문제가 있을 때 금방 해결책을 찾는 경험치가 될 수 있었다.
- 초급 프로젝트를 진행할 때는 api관련 기능 위주로 작업했다가 중급 프로젝트에 와서 컴포넌트 위주로 담당하게 되었다. 이 컴포넌트는 나만 사용하는 것이 아니니 사용성과 가독성, 그리고 요구사항에 직접적으로 드러나지 않는 사양도 고려해야 함을 더욱더 실감하게 했다.

# 후기
- 좀 더 가독성.... 가독성... 가독성...
- 고민이 생기면 오래 고민하지 말자! 내 고집으로 프로젝트 진도를 늦추지 말자!
- 말솜씨는 중요하다. 영독도 중요하다. 공부공부.
- 가장 기초적인 부분에서 답이 나올 때가 있다.

# 발표자에게
- 질문 사절에대한 강한 어필이 필요할 때입니다.!!!!!!!!!!!11

**Comment by caseV12** [comment]
# 맡은 부분
 - Input 공통 컴포넌트 만들었습니다.
 - 로그인, 회원가입 페이지, AuthContext 만들었습니다.
 - 찬주님께서 작성하신 useAsync 기반으로 useAxios 훅 만들었습니다.

# 이런 생각 많이 했어요
- 비제어 컴포넌트와 제어 컴포넌트에 대해 생각하며 불필요한 state를 피하는 것에 신경을 많이 썼습니다. 팀원들이 제가 작성한 부분을 읽고 재사용하는 데에 적은 리소스가 들도록 제 능력 내에서는 최대한 신경쓰고 노력해보았습니다.  

# 후기
### 이런 거 좋았다
- 경험도 있고 배울 점도 상당한 팀원 분들이랑 매칭이 되어 코드 보고 배울 수 있었습니다. 큰 도움이 되었습니다.
- 코드 리뷰를 처음 해봤는데, 다른 분들의 코드를 읽고 이해하는 것이 익숙치 않아 오랜 시간을 투자해야 하는 어려움이 있었지만 좋은 연습이 되었습니다.
- 다른 팀원분들께서 기본 셋팅과 기록 등 체계를 잘 잡아주셔서 프로젝트를 원활히 진행할 수 있었던 것 같습니다. 프로젝트의 진행에 대해서도 많이 배웠습니다.
- 타입스크립트를 쓰는 게 매우 심히 힘들었지만 많이 배우게 되었습니다.
### 이런 거 아쉽다
 - 개인적인 부분인데, 현실적으로 제 실력이 안 따라주는 상황에서 팀원분들처럼 좀 좋은 코드를 쓰고싶다는 마음이 생겨 작업 시간이 길어지고 쏟는 시간에 비해 전체적인 코드의 양에 기여를 많이 못했습니다. 팀에 독이 된 것 같습니다. 기능 구현을 우선적으로 하고 추후 리팩터링에 노력을 기울이는 것도 방법이였을 것 같습니다.(근데 성격 상 그게 안돼요...) 

# 발표자에게

**Comment by sung-eunji** [comment]
## 맡은부분
Chips#10, Card#41 컴포넌트 작성. TodoModal #82 초반작업.

## 이런 생각 많이 했어요
뉴런 사멸에 기여한 문제
: TodoModal은 맡기는 했으나 api를 post axios 작성까지만 하고 유저 데이터를 get 해오는 과정 및 전반적인 모달 구성계획에 난항을 겪다 sos를 청했습니다. 
를 해결한 방법
: 설연휴 기간에 저녁 8시에 팀미팅을 시작했는데 팀원분들이 새벽 4시까지 디스코드 채널톡 및 쉐어스크린과 vscode 라이브쉐어를 통해 모달 작성에 도움을 주셨습니다. 오마야 

Typescript 사용하면서 처음에는 타입에러 나는것이 많이 스트레스였는데, 이 에러가 오히려 제가 에러를 찾아 헤메는 시간을 줄여서 이정도라는 사실을 알았습니다. 
개인적으로는 실재 서비스구현에서 상태 관리에 따라 비용문제와 직결된다는 사실이 흥미로웠습니다. 

## 후기

### 1. 이런 거 아쉽다!
: 로그인페이지, 모달작성과 같이 처음 해보지만 팀프로젝트에서 도움받을 수 있는 기술구현들을 시도할 수 없었습니다. 제 능력부족으로 기술적인 기여도가 많이 없어서 많이 아쉽습니다. 

### 2. 이런 거 좋았다!
: 하드스킬은 많이 부족하지만 팀원들 간에 소통이 이루어질 때 질문 하고 답변을 들으면 그것을 제 언어로 다시 정리해 보기를 하면서 서로가 모르는 영역의 테두리를 잡아가는 과정을 보냈던 것 같습니다. 
기술적으로 저보다 잘하는 분들과 함께해서 아주 자잘한 과정들부터 큰 생각까지 많이 배웠습니다.

**Comment by BBoBBoM** [comment]
# 이건 꼭 넣고 싶다
- [next-themes]  : DarkMode 구성에 필요 간편한 테마 관리
- [clsx] 클래스 정의시 가독성 재사용성 구조화된 코드 작성에 도움
- 
- 텍스트/아이콘 버튼 공통 컴포넌트 
- LandingPage UI구현 다크모드 구현, Token보유시 router 동작 부분
- myPage profile 닉네임, 비밀번호 변경 구현
- dashboard/{dashboardId} page 기본 UI 구현, 컬럼추가/이름변경, 컬럼 삭제 기능구현, card목록 무한스크롤, 할일 생성 모달 생성시 재렌더 되지 않는 문제 해결
- SideMenu dashBoard 생성버튼 이벤트 구현 

# 이런 생각 많이 했어요
- 공통 컴포넌트를 수준높게 만들어 많은 부분에 사용할수 있도록 했으면 좋지 않았을까
- 코드가 지저분하고 본인만 알아 볼수있으면 곤란하니 가독성을 높이고 싶어 
- 다른 팀원/사람 코드를 많이 참고하고 읽고 받아들여야겠다 

# 후기
- pr리뷰를 작성해야하니 코드동작 부분 및 코드 작성 형태에 대해서 꼼꼼히 봐야했습니다 많은 부분 배우고 개선하고 이후 적용해보고 싶다는 생각을 많이 했습니다.
- 라이브 쉐어로 소통하고 같이 코드도 작성한 시간이 재미있고 유익했습니다.
- 적극적으로 프로젝트에 임하고 본인 작업의 진행상황에 대한 공유와 전파가 잘되었고 할일 부분도 세밀하게 분리하여 작업범위에 대한 명확성이 좋았으며 오류 문제 해결시  팀원들과 같이 해결하거나 해결 방향을 제시하는등의 커뮤니케이션이 잘 이루어졌습니다. 
- 요구 사항 이해오류로 잦은 수정부분, 공통 컴포넌트를 좀 더 폭넓고 사용하기 쉽게 구현하지 못한 부분의 아쉬움
- 효과적이고 가독성 높은 코드를 작성하지 못한 능력에 대한 아쉬움 
- 특정 문제 설명에 관한 언어표현의 어려움......... (말을 잘 못함) 
- typescript 사용 미숙으로 타입에러 다량 유발에 대한 미안함................................
- 
# 발표자에게
- 질문 사절에대한 강한 어필이 필요할 때입니다.!!!!!!!!!!!11
- 리뷰 많이 달아주시고 전체적인 지휘 효과적인 방법을 많이 생각해주신 찬주님 감사 인사 드렸으면 좋겠습니다 고생한 팀원에 대한 노고도 위로해주는 한마디 좋을거 같습니다.

**Comment by ckswnskfk** [comment]
# 이건 꼭 넣고 싶다
- useModal 훅. 개선할 사항이 있지만 사용하는 입장에서 간편하게 사용할 수 있도록 모달을 공통 모듈화 했습니다.
- login, 회원가입 페이지 제외한 공통 레이아웃

# 이런 생각 많이 했어요
- 재사용 가능한 컴포넌트를 작성하기 위해 공통적으로 많이 사용되는 코드를 모듈화 하려고 노력했습니다.
- 코드 베이스의 품질과 일관성을 위해 코드 리뷰를 꼼꼼히 하려고 노력했습니다.

# 후기
- 모나지 않은 팀원들이 잘 모여서 작은 불화 없이 프로젝트를 잘 진행했습니다.
- 나름대로 각자의 의견을 말할 수 있는 좋은 환경에서 개발해서 좋았습니다.
- 기능을 다 구현하지 못했고, 최적화에 거의 신경을 쓰지 못해서 아쉽습니다.
- nextjs 로 구축했지만 서버 사이드 렌더링을 적용하지 못한 점이 아쉽습니다. accessToken을 document.cookie 에라도 담아서 사용할 걸 그랬습니다.
- 서버 상태를 효율적으로 관리하지 못했습니다. context 를 좀 더 적극적으로 썼으면 하는 아쉬움이 있습니다.
- 타입스크립트도 좀 더 정확하고 유연하게 사용하고 싶다는 아쉬움이 있습니다.

# 발표자에게
-
